vignettes/databases.Rmd

```diff
@@ -0,0 +1,136 @@
+---
+title: "Use External Databases with shiny.telemetry"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Use External Databases with shiny.telemetry}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+The `{shiny.telemetry}` package can be used with any Shiny application and in this guide we will show how to use it with different databases backend.
+
+A requirement for this guide is to have a database server running and the necessary permissions to create a database, tables and insert data.
+
+The following databases are supported by `{shiny.telemetry}`:
+
+- PostgreSQL
+- MariaDB or MySQL
+- MS SQL Server
+- MongoDB
+- SQLite
+
+We provide example applications for each database backend with necessary R code to run both the application and the analytics server.
+This is further supported with a `docker-container.yml` to help users quickly setup and test the apps locally.
+It requires Docker (`docker compose up -d`) or Podman (`podman-compose up -d`) installed to run the containers.
+
+These applications are available under the `inst/examples/` folder or via the [GitHub link](https://github.com/Appsilon/shiny.telemetry/tree/main/inst/examples).
+
+## Create a data storage backend {.tabset}
+
+Each data storage backend will create the necessary tables _(or "Collection" in the case of MongoDB)_ with the respective schema when needed.
+
+The arguments to create an data storage instance vary, as different databases require their own options.
+However, once the data storage object is created, the read and write operations have the same API.
+
+Below you find chunks to create a data storage object for each supported database.
+
+### PostgreSQL
+
+```r
+data_storage <- DataStoragePostgreSQL$new(
+  user = "postgres",
+  password = "mysecretpassword",
```

Comment:
Here and below - please add a note about not storing password in code (and suggest env vars as a solution, with .Renviron as a default way of doing that in R).

Reply:
Explicitly used user and passwords from environment variables. 

Should we also add a note to server/port? or would that be overkill for this guide (not overkill for a production environment)

Alternatively, we could add a section with "How to protect credentials?" and say it again for users, but also server, port, database, ...